clients/vscode-hlasmplugin/src/hlasmExternalFiles.ts

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2023 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+
+import * as vscode from "vscode";
+import * as vscodelc from "vscode-languageclient";
+
+// This is a temporary demo implementation
+
+enum ExternalRequestType {
+    read_file = 'read_file',
+    read_directory = 'read_directory',
+}
+
+interface ExternalRequest {
+    id: number,
+    op: ExternalRequestType,
+    url: string,
+}
+
+interface ExternalReadFileResponse {
+    id: number,
+    data: string,
+}
+interface ExternalReadDirectoryResponse {
+    id: number,
+    data: string[],
+}
+interface ExternalErrorResponse {
+    id: number,
+    error: {
+        code: number,
+        msg: string,
+    },
+}
+
+function generateFileContent(uriPath: string) {
+    return `.*
+         MACRO
+         ${uriPath.substring(uriPath.length - 4)}
+         MEND`
+}
+
+const magicSchema = 'hlasm-external';
```

Comment:
I agree, but this is really just a demo implementation, so that there is a way to validate the concept. Basically, this whole file will be rewritten with the actual implementation in the future...